benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/dataloader/DataLoader.java

```diff
@@ -270,7 +282,125 @@ public long loadData() throws Exception {
     return totalRowCount;
   }
 
-  long loadTable(AbstractRowProducer rowProducer) throws SQLException, IOException {
+  long loadTable(AbstractRowProducer rowProducer)
+      throws RuntimeException, SQLException, IOException {
+    if (isGoogleSQL) {
+      return loadTableInGSQL(rowProducer);
+    }
+    return loadTableInPG(rowProducer);
+  }
+
+  String unquote(String input) {
+    if (input.length() >= 2 && input.startsWith("'") && input.endsWith("'")) {
+      return input.substring(1, input.length() - 1);
+    }
+    return input;
+  }
+
+  void setMutationColumn(WriteBuilder writeBuilder, String name, Object value)
+      throws RuntimeException {
+    ValueBinder valueBinder = writeBuilder.set(name);
+    if (value instanceof Boolean) {
+      valueBinder.to((Boolean) value);
+    } else if (value instanceof Long) {
+      valueBinder.to((Long) value);
+    } else if (value instanceof Float) {
+      valueBinder.to((Float) value);
+    } else if (value instanceof Double) {
+      valueBinder.to((Double) value);
+    } else if (value instanceof BigDecimal) {
+      valueBinder.to((BigDecimal) value);
+    } else if (value instanceof Timestamp) {
+      Timestamp timestamp = (Timestamp) value;
+      if (timestamp.equals(Timestamp.MIN_VALUE)) {
+        // We use Timestamp.MIN_VALUE to represent NULL as ImmutableList does not permit
+        // NULLs.
+        valueBinder.to((Timestamp) null);
+      } else {
+        valueBinder.to(timestamp);
+      }
+    } else if (value instanceof String) {
+      // Unquote the string because the generated strings are wrapped with single
+      // quotes.
+      String stringValue = unquote((String) value);
+      if (stringValue.isEmpty()) {
+        valueBinder.to((String) null);
+      } else {
+        valueBinder.to(stringValue);
+      }
+    } else {
+      throw new RuntimeException(String.format("Unknown value for column %s: %s", name, value));
+    }
+  }
+
+  long loadTableInGSQL(AbstractRowProducer rowProducer)
+      throws RuntimeException, SQLException, IOException {
+    long count = 0;
+    try (Connection connection = createConnection();
+        Statement statement = connection.createStatement()) {
+      // Unwrap the Cloud Spanner specific interface to be able to access custom
+      // methods.
+      CloudSpannerJdbcConnection spannerConnection =
+          connection.unwrap(CloudSpannerJdbcConnection.class);
+      spannerConnection.setAutoCommit(false);
+
+      List<Mutation> mutations = new ArrayList<>(Math.round(MUTATION_LIMIT_PER_COMMIT / 4));
```

Comment:
nit: Can you add a comment for why you do a 'divide limit by 4' here?

Reply:
Added a comment.